server/personality/personality.controller.ts

```diff
@@ -22,13 +22,19 @@ import { IsPublic } from "../auth/decorators/is-public.decorator";
 import { TargetModel } from "../history/schema/history.schema";
 import { BaseRequest } from "../types";
 import { ApiTags } from "@nestjs/swagger";
+import { ConfigService } from "@nestjs/config";
+import { CaptchaService } from "../captcha/captcha.service";
+import { HistoryService } from "../history/history.service";
 
 @Controller()
 export class PersonalityController {
     private readonly logger = new Logger("PersonalityController");
     constructor(
         private personalityService: PersonalityService,
-        private viewService: ViewService
+        private viewService: ViewService,
+        private configService: ConfigService,
+        private captchaService: CaptchaService,
+        private historyService: HistoryService
     ) {}
 
     @IsPublic()
```

Comment:
chore: create a task to enhance RBAC for these public routes. The isHidden attribute should only be used by admins.